soroban-env-host/src/native_contract/token/error.rs

```diff
@@ -1,35 +1,23 @@
-use crate::host::HostError;
-use soroban_env_common::{ConversionError, Status};
+use crate::{events::DebugError, host::HostError, Host};
+use soroban_env_common::Status;
 
 #[derive(Debug)]
-pub enum Error {
-    HostError(HostError),
-    ContractError,
-}
-
-impl From<Status> for Error {
-    fn from(s: Status) -> Self {
-        Error::HostError(s.into())
-    }
-}
+pub enum ContractError {
+    InternalError = 1,
+    OperationNotSupportedError = 2,
+    AlreadyInitializedError = 3,
 
-impl From<ConversionError> for Error {
-    fn from(e: ConversionError) -> Self {
-        Error::HostError(e.into())
-    }
-}
+    UnauthorizedError = 4,
+    AuthenticationError = 5,
+    NonceError = 6,
+    SignatureError = 7,
 
-impl From<HostError> for Error {
-    fn from(e: HostError) -> Self {
-        Error::HostError(e)
-    }
+    NegativeAmountError = 8,
+    AllowanceError = 9,
+    BalanceError = 10,
+    BalanceFrozenError = 11,
 }
 
-impl From<Error> for HostError {
-    fn from(e: Error) -> Self {
-        match e {
-            Error::HostError(he) => he,
-            Error::ContractError => crate::xdr::ScUnknownErrorCode::General.into(),
-        }
-    }
+pub fn contract_err(host: &Host, err: ContractError, msg: &'static str) -> HostError {
+    host.err(DebugError::new(Status::from_contract_error(err as u32)).msg(msg))
```

Comment:
I don't think you need this function. This is basically exactly what the `err_status_msg` (https://github.com/stellar/rs-soroban-env/blob/main/soroban-env-host/src/host/err_helper.rs#L39) is for.

Reply:
Thanks, I've only somehow noticed message-only and status-only fns. Changed code to use `err_status_msg`.